src/inference/src/dev/core_impl.cpp

```diff
@@ -1397,42 +1397,97 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::load_model_from_cache(
     struct HeaderException {};
 
     OPENVINO_ASSERT(cacheContent.cacheManager != nullptr);
+    bool use_mmap = plugin.supports_model_caching_with_mmap();
     try {
-        cacheContent.cacheManager->read_cache_entry(cacheContent.blobId, [&](std::istream& networkStream) {
-            OV_ITT_SCOPE(FIRST_INFERENCE,
-                         ov::itt::domains::LoadTime,
-                         "Core::load_model_from_cache::ReadStreamAndImport");
+        bool mmap_failed = false;
```

Comment:
What the purpose to have a fallback on read in case if `mmap` failed? I think there are no obvious reasons `mmap` should fail, but if it will happen, the FIL will be affected, and it will be hard to detect